include/boost/smart_ptr/detail/shared_count.hpp

```diff
@@ -661,7 +661,7 @@ inline shared_count::shared_count( weak_count const & r, sp_nothrow_tag ) BOOST_
 #pragma GCC diagnostic pop
 #endif
 
-#ifdef __BORLANDC__
+#ifdef BOOST_BORLANDC
```

Comment:
This change doesn't agree with the previous one; this uses BOOST_BORLANDC, the other does not.

Reply:
If config.hpp is not included or not yet included we can not use BOOST_BORLANDC but instead must use the long form of __BORLANDC__ and !__clang__.

Reply:
If you use one condition for the `# pragma warn -8027`, you must use the same condition for the subsequent `# pragma warn .8027`.

Reply:
Done !